### ISSUE TITLE
how  to provide accessible name for country code form field

### ISSUE THREAD

**Issue by srikanthgsm** [hbb20/CountryCodePickerProject#460]
how  to provide accessible name for country code form field? 
as of now talkback only announces the value country code. I need to provide custom accessible name for country code form field.

**Comment by hbb20** [comment]
Thanks you for reporting. I will have to take a look at it.

**Comment by srikanthgsm** [comment]
Hi team, did you get chance to look into it. Expecting your fix ASAP.

**Comment by hbb20** [comment]
It's been a crazy week. Sorry about that. If you can create a PR , that
would be great I'll be happy to merge and release new version. Otherwise
please explain little more about what exactly you are looking for since
accessibility is not my strong hold yet.

**Comment by srikanthgsm** [comment]
When I enable talkback and navigate to edit phone number screen..., I have country code picker. In which country code is prefilled based on the region where the user is located. while the country code field(_country code field doesn't have label as per design_) is focused, talkback announced the value country code. The expected behavior is, **talkback should announce ("Country code is +1) but now the country code pick announces simply (+1).** 
The differently abled users cannot understand what is +1 when talkback announces, if we provide proper value as ("Country code is +1<<respective_country_code>>). Then it would be useful.
**Because of this issue our app fails for accessibility standards.**

Thanks

**Comment by hbb20** [comment]
Thanks for the feedback. I will look into how can I do this and will do
tonight after office. I don't want you to hold any longer because of this.
Meanwhile if you know and can share how can I do that that would be greatly
appreciated.

**Comment by srikanthgsm** [comment]
I could see countrypicker extends Relativelayout. with this I have added **ContentDescription**. but **ContentDescription** is not read by talkback, rather it reads the value of the formfield. below is my xml.

<com.hbb20.CountryCodePicker
                    android:id="@+id/country_picker"
                    android:layout_width="wrap_content"
                    android:layout_height="wrap_content"
                    android:layout_marginLeft="40dp"
                    android:layout_marginTop="20dp"
                    android:layout_marginRight="10dp"
                    **android:contentDescription="country code is +1"**
                    android:background="@drawable/edit_text_book_border"
                    android:gravity="center|center_vertical"
                    android:includeFontPadding="false"
                    android:minHeight="65dp"
                    app:ccpDialog_backgroundColor="@color/colorWhite"
                    app:ccpDialog_showCloseIcon="true"
                    app:ccpDialog_textColor="@color/colorPurple"
                    app:ccp_areaCodeDetectedCountry="true"
                    app:ccp_autoDetectCountry="true"
                    app:ccp_countryPreference="@string/drop_down_country_pref"
                    app:ccp_showNameCode="false" />

Hope this will help.

**Comment by hbb20** [comment]
Thanks for heads up. Will get into this and get back to you.

**Comment by hbb20** [comment]
Hey, 

I have released new version 2.4.5 which should have better talkback content. Default will be `country.name + " phone code is +" + country.phoneCode;`

You can also override this by setting your own `CCPTalkBackTextProvider`. You can find an example [here](https://github.com/hbb20/CountryCodePickerProject/blob/master/app/src/main/java/in/hbb20/countrycodepickerproject/IntroductionFragment.kt#L47).

Please let me know if that works for you.

**Comment by hbb20** [comment]
@srikanthgsm ☝🏼

**Comment by srikanthgsm** [comment]
Will try and update you. 
Thanks

**Comment by srikanthgsm** [comment]
i'm getting below error while gradle sync. but with 2.4.4 compiling, only 2.4.5 i'm getting error.

Could not determine the dependencies of task ':app:compileProdDebugJavaWithJavac'.
> Could not resolve all task dependencies for configuration ':app:prodDebugCompileClasspath'.
   > Could not find com.hbb20:ccp:2.4.5.
     Required by:
         project :app

**Comment by hbb20** [comment]
oops.. build failed.. try again please.

**Comment by srikanthgsm** [comment]
Thanks Harsh, build successful. The fix is working.
